packages/kuma-gui/src/app/application/index.ts

```diff
@@ -83,6 +84,44 @@ const $ = {
   i18n: token<ReturnType<typeof I18n>>('i18n'),
   enUs: token('i18n.locale.enUs'),
 }
+
+const addModule = (item: RouteRecordRaw, parent?: RouteRecordRaw) => {
+  item.meta = {
+    ...(item.meta ?? {}),
+  }
+  if (typeof parent?.meta?.module !== 'undefined') {
+    item.meta.module = parent.meta.module
+  }
+}
+const addPath = (item: RouteRecordRaw, parent?: RouteRecordRaw) => {
+  item.meta = {
+    ...(item.meta ?? {}),
+  }
+  if (typeof parent?.meta?.path !== 'undefined') {
+    const path = String(parent.meta.path) ?? ''
+    item.meta.path = `${path}${path.length > 0 ? '.' : ''}${String(item.name)}`
+  }
+}
+const addRouteName = (item: RouteRecordRaw) => {
+  if (typeof item.name !== 'undefined') {
+    const props = ((props) => {
+      switch (true) {
+        case typeof props === 'function':
+          return props
+        case typeof props === 'undefined':
+          return () => ({})
+        default:
+          return () => props
+      }
+    })(item.props)
+    item.props = (...args) => {
+      return {
+        ...props(...args),
+        routeName: item.name,
+      }
+    }
+  }
+}
```

Comment:
Just a nit 🙂 would turn around the `if`-condition and exit early

Reply:
Oh! Sorry completely missed this, this is done now, lmk!